include/oneapi/dpl/internal/dynamic_selection.h

```diff
@@ -71,6 +72,18 @@ namespace experimental {
     return std::forward<Handle>(h).wait_for_all();
   }
 
+  template<typename Handle>
+  auto wait_for_all(std::list<Handle> l) {
```

Comment:
Having std::list to differentiate between wait(Handle &h) v/s wait(HandleList &l) since there is no way to specify a single element vs a list without std::concepts